src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,368 @@
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private void startWithOptions(final JSONObject jsonOptions, final CallbackContext callbackContext) {
+    SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
+      try {
+        if (jsonOptions.has("dsn") && jsonOptions.getString("dsn") != null) {
           String dsn = jsonOptions.getString("dsn");
+          logger.info(String.format("Starting with DSN: '%s'", dsn));
+          options.setDsn(dsn);
+        } else {
+          // SentryAndroid needs an empty string fallback for the dsn.
+          options.setDsn("");
+        }
+        if (jsonOptions.has("debug") && jsonOptions.getBoolean("debug")) {
+          options.setDebug(true);
+        }
+        if (jsonOptions.has("maxBreadcrumbs")) {
+          options.setMaxBreadcrumbs(jsonOptions.getInt("maxBreadcrumbs"));
+        }
+        if (jsonOptions.has("environment") && jsonOptions.getString("environment") != null) {
+          options.setEnvironment(jsonOptions.getString("environment"));
+        }
+        if (jsonOptions.has("release") && jsonOptions.getString("release") != null) {
+          options.setRelease(jsonOptions.getString("release"));
+        }
+        if (jsonOptions.has("dist") && jsonOptions.getString("dist") != null) {
+          options.setDist(jsonOptions.getString("dist"));
+        }
+        if (jsonOptions.has("enableAutoSessionTracking")) {
+          options.setEnableSessionTracking(jsonOptions.getBoolean("enableAutoSessionTracking"));
+        }
+        if (jsonOptions.has("sessionTrackingIntervalMillis")) {
+          options.setSessionTrackingIntervalMillis(jsonOptions.getInt("sessionTrackingIntervalMillis"));
+        }
+        if (jsonOptions.has("enableNdkScopeSync")) {
+          options.setEnableScopeSync(jsonOptions.getBoolean("enableNdkScopeSync"));
+        }
+        if (jsonOptions.has("attachStacktrace")) {
+          options.setAttachStacktrace(jsonOptions.getBoolean("attachStacktrace"));
+        }
+        if (jsonOptions.has("attachThreads")) {
+          // JS use top level stacktraces and android attaches Threads which
+          // hides them so by default we hide.
+          options.setAttachThreads(jsonOptions.getBoolean("attachThreads"));
+        }
 
-          SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
-            options.setDsn(dsn);
-          });
+        if (jsonOptions.has("enableNativeCrashHandling") && !jsonOptions.getBoolean("enableNativeCrashHandling")) {
+          final List<Integration> integrations = options.getIntegrations();
+          for (final Integration integration : integrations) {
+            if (integration instanceof UncaughtExceptionHandlerIntegration || integration instanceof AnrIntegration || integration instanceof NdkIntegration) {
+              integrations.remove(integration);
+            }
+          }
```

Comment:
instead of doing this, you could use the boolean flags
https://github.com/getsentry/sentry-dart/blob/main/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt#L97-L105